html/browsers/browsing-the-web/scroll-to-fragid/scroll-frag-non-utf8-encoded-document.html

```diff
@@ -0,0 +1,42 @@
+<!doctype html>
+<title>Fragment Navigation: fragment id should not be found in non UTF8 document</title>
+<meta name=timeout content=long>
+<meta http-equiv="Content-Type" content="text/html; charset=gbk"/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<div></div>
+<div id="&#x586f" style="position:absolute; top:100px;"></div>
+<div style="height:200em;"></div>
+<script>
+var steps = [{
```

Comment:
It looks like this style is copied from sibling tests which have more than one step. With just one step, I find it quite hard to follow with all the indirection.

Can you try flattening it into a simple single-page test, be something like this?
```js
setup({ single_test: true });
assert_equals(location.hash, "", "Page must be loaded with no hash");
location.hash = '%89g';
setTimeout(() => {
  assert_equals( document.scrollingElement.scrollTop, 0 );
  done();
}, 1);
```

I think that's all the test is doing, but I haven't tested my suggestion.

Reply:
Thanks! I tested it and it works, so I pushed that new version. Initially I thought perhaps we will add more subtests so I kept the multiple tests approach, but I guess we can introduce it later if needed.